lib/hutch/broker.rb

```diff
@@ -125,7 +125,14 @@ def open_channel!
     end
 
     def declare_exchange(ch = channel)
-      channel_broker.declare_exchange(ch)
+      exchange_name = @config[:mq_exchange]
```

Comment:
As we call `ChannelBroker#declare_exchange` from within the `ChannelBroker`, these changes (adding `exchange_type` and using the `Adapter.new_exchange`) should be copied through there too.